components/draw/GeometryMode.svelte

```diff
@@ -0,0 +1,169 @@
+<script>
+  import { map, gjScheme, currentHover } from "../../stores.js";
+
+  const thisMode = "edit-geometry";
+
+  export let mode;
+  export let routeSnapper;
+  export let snapTool;
+  export let drawControls;
+
+  export function start() {}
+  export function stop() {
+    if (currentlyEditing) {
+      // We could've been editing anything; just handle all possibilities
+      drawControls.changeMode("static");
+      drawControls.deleteAll();
+
+      routeSnapper.stop();
+
+      gjScheme.update((gj) => {
+        let feature = gj.features.find((f) => f.id == currentlyEditing);
+        delete feature.properties.hide_while_editing;
+        return gj;
+      });
+    }
+
+    currentlyEditing = null;
+    currentHover.set(null);
+  }
+
+  // This is different than the stores.js one!
+  let currentlyEditing = null;
```

Comment:
Why do we use a local `currentlyEditing` here rather than the one from `stores.js`?

Reply:
I'll change the comment. We need better terminology; "editing" now means "editing geometry" or "editing properties". This is "editing geometry", the one in the store is for properties.